java/code/src/com/suse/manager/webui/services/impl/SaltSSHService.java

```diff
@@ -559,9 +568,23 @@ private boolean addSaltSSHMinionsFromDb(SaltRoster roster) {
                         roster,
                         parameters.isIgnoreHostKeys(),
                         isSudoUser(parameters.getUser()));
+        tmpKeyFileAbsolutePath.ifPresent(path -> cleanUpTempKeyFile(path));
```

Comment:
To be on the safe side, Ithe call to `callSyncSSHInternal` could be wrapped in a `try { ... } finally { ... }` block and the cleanup of the key moved inside `finally { ... }`.

Reply:
Great catch, fixed!